atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java

```diff
@@ -1201,6 +1204,101 @@ public void testGetSortedColumnsReadWriteConflictAtEndOfBatch() {
         assertThatThrownBy(t1::commit).isInstanceOf(TransactionSerializableConflictException.class);
     }
 
+    @Test
+    public void testGetSortedColumnsLocalWritesBeforeAreReadAndDoNotTriggerConflict() {
+        List<byte[]> rows = generateRows(5);
+        List<Cell> cellsWrittenOriginally = generateCells(rows, generateColumns(5));
+        writeCells(cellsWrittenOriginally);
+
+        Transaction t1 = startTransactionWithSerializableConflictChecking();
+        byte[] newValue1 = PtBytes.toBytes("find a way");
+        byte[] newValue2 = PtBytes.toBytes("persevere to the end");
+        t1.put(TEST_TABLE, ImmutableMap.of(cellsWrittenOriginally.get(0), newValue1));
+        t1.put(TEST_TABLE, ImmutableMap.of(cellsWrittenOriginally.get(17), newValue2));
+        Iterator<Map.Entry<Cell, byte[]>> sortedColumns = t1.getSortedColumns(
+                TEST_TABLE,
+                rows,
+                BatchColumnRangeSelection.create(
+                        PtBytes.EMPTY_BYTE_ARRAY, PtBytes.EMPTY_BYTE_ARRAY, DEFAULT_BATCH_HINT));
+
+        List<Map.Entry<Cell, byte[]>> sortedColumnValues =
+                Streams.stream(sortedColumns).collect(Collectors.toList());
+        List<Cell> sortedCellsRead =
+                sortedColumnValues.stream().map(Entry::getKey).collect(Collectors.toList());
+        sanityCheckOnSortedCells(rows, sortedCellsRead, cellsWrittenOriginally);
+        assertThat(sortedColumnValues)
+                .contains(
+                        Maps.immutableEntry(cellsWrittenOriginally.get(0), newValue1),
+                        Maps.immutableEntry(cellsWrittenOriginally.get(17), newValue2));
+        t1.commit();
```

Comment:
nit: given the name of the test - 
```suggestion
        assertThatCode(t1::commit).doesNotThrowAnyException();
```